main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java

```diff
@@ -379,6 +400,114 @@ private void loadFeedInfo(GtfsTableContainer<GtfsFeedInfo> feedTable) {
+        for (GtfsCalendar calendar : calendars) {
+          if (calendar.serviceId().equals(serviceId)) {
+            LocalDate startDate = calendar.startDate().getLocalDate();
+            LocalDate endDate = calendar.endDate().getLocalDate();
+
+            if (startDate != null || endDate != null) {
+              if (earliestStartDate == null || startDate.isBefore(earliestStartDate)) {
+                earliestStartDate = startDate;
+              }
+              if (latestEndDate == null || endDate.isAfter(latestEndDate)) {
+                latestEndDate = endDate;
+              }
+            }
+          }
+        }
+      }
+    } else if ((calendarDateTable != null) && (calendarTable == null)) {
+      // When only calendar_dates.txt is used
+      List<GtfsCalendarDate> calendarDates = calendarDateTable.getEntities();
+      for (GtfsTrip trip : trips) {
+        String serviceId = trip.serviceId();
+        for (GtfsCalendarDate calendarDate : calendarDates) {
+          if (calendarDate.serviceId().equals(serviceId)) {
+            LocalDate date = calendarDate.date().getLocalDate();
+            if (date != null) {
+              if (earliestStartDate == null || date.isBefore(earliestStartDate)) {
+                earliestStartDate = date;
+              }
+              if (latestEndDate == null || date.isAfter(latestEndDate)) {
+                latestEndDate = date;
+              }
+            }
+          }
+        }
+      }
+    } else if ((calendarTable != null) && (calendarDateTable != null)) {
+      // When both calendars.txt and calendar_dates.txt are used
+      Map<String, ServicePeriod> servicePeriods =
+          CalendarUtil.buildServicePeriodMap(
+              (GtfsCalendarTableContainer) calendarTable,
+              (GtfsCalendarDateTableContainer) calendarDateTable);
+      List<LocalDate> removedDates = new ArrayList<>();
+      for (GtfsTrip trip : trips) {
+        String serviceId = trip.serviceId();
+        ServicePeriod servicePeriod = servicePeriods.get(serviceId);
+        LocalDate startDate = servicePeriod.getServiceStart();
+        LocalDate endDate = servicePeriod.getServiceEnd();
+
+        if (startDate != null && endDate != null) {
+          if (earliestStartDate == null || startDate.isBefore(earliestStartDate)) {
+            earliestStartDate = startDate;
+          }
+          if (latestEndDate == null || endDate.isAfter(latestEndDate)) {
+            latestEndDate = endDate;
+          }
+        }
+        removedDates.addAll(servicePeriod.getRemovedDays());
+      }
+
+      for (LocalDate date : removedDates) {
+        if (date.isEqual(earliestStartDate)) {
+          earliestStartDate = date.plusDays(1);
+        }
+        if (date.isEqual(latestEndDate)) {
+          latestEndDate = date.minusDays(1);
+        }
+      }
+    }
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMMM d, yyyy");
+    if ((earliestStartDate == null) && (latestEndDate == null)
```

Comment:
Won't this generate an NPE if earliestStartDate is null and latestEndDate is not null?